### PR TITLE
Add NFTs to airdrop REST APIs

### LINKED ISSUE #9378: HIP-904 Add NFTs to Airdrop REST API response
### Problem

The Outstanding and Pending airdrops endpoints currently only return fungible tokens.

### Solution

-Add nfts to the response and paginate based on sender/receiver, token.id and serialnumber.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:

Add NFT support to the outstanding and pending airdrop REST APIs

**Related issue(s)**:

Fixes #9378

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/TokenAirdropsControllerTest.java [hunk 1]
```diff
@@ -297,76 +439,333 @@ void allParameters() {
...
+            nextParams =
+                    "?limit=1&receiver.id=lt:0.0.4&receiver.id=gte:0.0.2&token.id=lte:0.0.10&token.id=gte:0.0.10&serialnumber=gt:11";
+            // Then
+            assertThat(result).isEqualTo(getExpectedResponse(List.of(airdropNft2), baseLink + nextParams));
+
+            // When
+            result = restClient.get().uri(nextParams, entity.getId()).retrieve().body(TokenAirdropsResponse.class);
+            // Serial number is removed from the next link
+            nextParams = "?limit=1&receiver.id=lt:0.0.4&receiver.id=gte:0.0.3&token.id=lte:0.0.10&token.id=gt:0.0.5";
+
```
Comment (edwin-greene) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/TokenAirdropsControllerTest.java:582]: This section tests the situation where the serial number needs to be removed from the link in order to progress to the next airdrop.

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/TokenAirdropsController.java [hunk 2]
```diff
@@ -113,9 +120,21 @@ private TokenAirdropsResponse processRequest(
...
+        var sort = getSort(airdrops, order, primarySortField);
         var pageable = PageRequest.of(0, limit, sort);
         var links = linkFactory.create(airdrops, pageable, EXTRACTOR);
         return new TokenAirdropsResponse().airdrops(airdrops).links(links);
     }
+
+    private Sort getSort(List<TokenAirdrop> airdrops, Sort.Direction order, String primarySortField) {
+        if (!airdrops.isEmpty()) {
+            var lastSerial = airdrops.getLast().getSerialNumber();
+            if (lastSerial == null || lastSerial == 0L) {
```
Comment (kselveliev) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/TokenAirdropsController.java:132]: nit: We could extract the 0L as a constant

**Review by kselveliev — DISMISSED** [review]

**Review by nickeynikolovv — APPROVED** [review]
LGTM

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/TokenAirdropsController.java (hunk 2 above)
Comment (edwin-greene) [reply hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/TokenAirdropsController.java]: Thanks I have added that.

**Review by steven-sheehy — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**